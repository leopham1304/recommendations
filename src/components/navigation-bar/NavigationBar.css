.nav-container {
  display: flex;
  justify-content: space-between;
  padding-top: 1vw;
  padding-bottom: 1vw;
  padding-left: 1vw;
}

.airbnb-logo {
  width: 30px;
}

.nav-menu {
  color: #fff;
  display: flex;
  align-items: center;
}

.nav-menu a {
  text-decoration: none;
  color: #fff;
  padding-right: 3vw;
  font-size: 14px;
  font-family: "Roboto", sans-serif;
}
