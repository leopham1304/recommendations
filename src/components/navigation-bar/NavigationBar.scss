.nav-container {
  display: flex;
  justify-content: space-between;
  padding-left: 1vw;
  align-items: center;
  height: 50px;

  .airbnb-logo {
    width: 30px;
  }

  .nav-menu {
    color: #fff;
    font-weight: 600;
    display: flex;
    align-items: center;
    span {
      text-decoration: none;
      color: #fff;
      padding-right: 3vw;
      font-size: 14px;
      cursor: pointer;
    }
  }

  @media screen and (max-width: 420px) {
    .nav-menu {
      flex-direction: column;
      margin-top: 50px;
    }
  }
}

.toggle {
  position: relative;
  padding: 0 4px;
  display: flex;
  align-items: center;

  input[type="checkbox"].checkbox {
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    width: 38px;
    height: 18px;
    background: #555;
    border-radius: 10px;
    cursor: pointer;
    outline: none;
    background-image: url(../../assets/images/night.svg);
    background-repeat: no-repeat;
    background-size: 30%;
    background-position: 5px center;
  }

  input[type="checkbox"].checkbox:checked + label {
    left: 30px;
  }

  input[type="checkbox"].checkbox:not(:checked) {
    background-image: url(../../assets/images/sunny.svg);
    background-repeat: no-repeat;
    background-size: 30%;
    background-position: 22px center;
  }

  input[type="checkbox"].checkbox + label {
    display: inline-block;
    width: 18px;
    height: 18px;
    border-radius: 50%;
    transition: all 0.3s ease;
    cursor: pointer;
    position: absolute;
    left: 8px;
    background-color: #fff;
  }
}
