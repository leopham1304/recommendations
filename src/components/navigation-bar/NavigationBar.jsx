import React, { memo } from "react";
import PropTypes from "prop-types";
import "./NavigationBar.scss";

export const NavigationBar = ({ setDarkMode, darkMode }) => {
  return (
    <nav className="nav-container">
      <div>
        <img
          className="airbnb-logo"
          src={require("../../assets/images/airbnb.svg")}
          alt="AirBnB Logo"
        />
      </div>
      <div className="nav-menu">
        <span className="toggle">
          <input
            checked={darkMode}
            onChange={() => setDarkMode((prevMode) => !prevMode)}
            id="checkbox"
            className="checkbox"
            type="checkbox"
          />
          <label htmlFor="checkbox" />
        </span>
        <span>Become a host</span>
        <span>Help</span>
        <span>Sign up</span>
        <span>Log in</span>
      </div>
    </nav>
  );
};

NavigationBar.propTypes = {
  darkMode: PropTypes.bool.isRequired,
  setDarkMode: PropTypes.func.isRequired,
};

export default memo(NavigationBar);
