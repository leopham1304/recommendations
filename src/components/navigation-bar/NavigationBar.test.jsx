import React from "react";
import NavigationBar from "./NavigationBar";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";

const mockSetDarkMode = jest.fn();

describe("NavigationBar", () => {
  const wrapper = mount(
    <NavigationBar darkMode={false} setDarkMode={mockSetDarkMode} />
  );

  it("should render the NavigationBar Component correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should bind the onChange of input correctly", () => {
    wrapper.find("input").simulate("change", {});
    expect(mockSetDarkMode).toBeCalled();
  });
});
