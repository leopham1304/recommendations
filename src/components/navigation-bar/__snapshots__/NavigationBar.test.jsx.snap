// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar should render the NavigationBar Component correctly 1`] = `
<Memo(NavigationBar)
  darkMode={false}
  setDarkMode={[MockFunction]}
>
  <nav
    className="nav-container"
  >
    <div>
      <img
        alt="AirBnB Logo"
        className="airbnb-logo"
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "airbnb.svg",
          }
        }
      />
    </div>
    <div
      className="nav-menu"
    >
      <span
        className="toggle"
      >
        <input
          checked={false}
          className="checkbox"
          id="checkbox"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="checkbox"
        />
      </span>
      <span>
        Become a host
      </span>
      <span>
        Help
      </span>
      <span>
        Sign up
      </span>
      <span>
        Log in
      </span>
    </div>
  </nav>
</Memo(NavigationBar)>
`;
